library/threading.c

```diff
@@ -30,6 +30,9 @@
 #include "mbedtls/threading.h"
 
 #if defined(MBEDTLS_THREADING_PTHREAD)
```

Comment:
Placing the code here, makes it dependent on `MBEDTLS_THREADING_PTHREAD` being defined which seems unnecessary.

If `MBEDTLS_THREADING_C` is defined and the target platform is Windows, I think that's sufficient.